### PR TITLE
[ISSUE-2442][linkis-basedata-manager] engine config template manage

### PR DESCRIPTION
**Pull request by ichenfeiyang**
### What is the purpose of the change

At present, the new engine needs to manually enter sql to prepare the engine parameters. This submission is the basic data management module, and the related interfaces for the configuration management of the engine parameters are added:
1. Query the corresponding configuration item template according to the selected application and engine
/api/rest_j/v1/configuration/getConfigurationTemplateByAppName
2. Add/modify configuration item template interface
/api/rest_j/v1/basedata-manager/configuration-template/save
3. Delete configuration item template interface
/api/rest_j/v1/basedata-manager/configuration-template/{keyId}

目前新增引擎需要手动录入sql进行引擎参数准备，该次提交是基础数据管理模块，新增引擎参数配置管理的相关接口：
1.根据选中的应用和引擎查询对应的配置项模板
/api/rest_j/v1/configuration/getConfigurationTemplateByAppName
2.新增/修改配置项模板接口
/api/rest_j/v1/basedata-manager/configuration-template/save
3. 删除配置项模板接口
/api/rest_j/v1/basedata-manager/configuration-template/{keyId}

### Related issues/PRs

Related issues: #2442
Related pr:#591

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Review by jacktao007 — APPROVED** [review]
LGTM.

**Review by jacktao007 — DISMISSED** [review]
LGTM.

File: linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/restful/ConfigurationTemplateRestfulApi.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+@RequestMapping(path = "/basedata-manager/configuration-template")
+public class ConfigurationTemplateRestfulApi {
+
+    @Autowired
+    ConfigurationTemplateService configurationTemplateService;
+
+    @ApiOperation(value = "save", notes = "save a configuration template", httpMethod = "POST")
+    @RequestMapping(path = "/save", method = RequestMethod.POST)
+    public Message add(@RequestBody ConfigurationTemplateSaveRequest request) {
+        if (Objects.isNull(request) || StringUtils.isEmpty(request.getEngineLabelId()) || StringUtils.isEmpty(request.getKey()) ||
```
Comment (casionone) [inline linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/restful/ConfigurationTemplateRestfulApi.java:48]: need add audit log for restful api such as:
```
  String userName=ModuleUserUtils.getOperationUser(request, "Save a configuration template, engineLabelId :" + request.getEngineLabelId());
```
Only administrators have permissions, and role judgments need to be added

File: linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/service/impl/ConfigurationTemplateServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import javax.annotation.Resource;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Lists;
+
```
Comment (casionone) [inline linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/service/impl/ConfigurationTemplateServiceImpl.java:44]: recommended to add some key logs

**Comment by casionone** [comment]
Need to supplement junit test

**Pull request closed without merging**